app/api/organizations/[id]/events/[eventId]/speakers/[speakerId]/router.ts

```diff
@@ -0,0 +1,22 @@
+import { NextResponse } from "next/server";
+import SpeakerController from "@/server/controller/speaker";
+export async function GET(
+  request: Request,
+  {
+    params,
+  }: {
+    params: { id: string; eventId: string; speakerId: string };
+  }
+) {
+  const speakerController = new SpeakerController();
+  try {
+    const data = await speakerController.getSpeaker(
+      params.speakerId,
+      params.eventId
+    );
+    return NextResponse.json(data);
+  } catch (e) {
+    console.log(e);
```

Comment:
i think console.error is very specific. this catch is handling all possible exceptions, for instance if the search query does not find an item with the request id it will throw an exception. This is an overall bad implementation of error handling that should be rewritten